### ISSUE TITLE
Util.Sleep() causing spike in CPU consumption

### ISSUE THREAD

**Issue by shinkarmv** [aerospike/aerospike-client-csharp#35]
While inserting the record, we have initialized the WritePolicy and set the - sleepBetweenRetries = 50

The below screenshot is stack trace from Dump analysis
![image](https://user-images.githubusercontent.com/31374791/36337767-2785be28-13c4-11e8-9a3c-23ce75c60d8a.png)

Gone through the code and found that Util.Sleep() internally calls - Thread.Sleep()

![image](https://user-images.githubusercontent.com/31374791/36337858-7e57db08-13c6-11e8-8440-8c99516a342a.png)

It looks like, once thread goes into sleep mode it unable wake up after sleep duration is expired.

Regards,
Mukesh

**Comment by BrianNichols** [comment]
Thread.Sleep() does not directly cause a spike in CPU consumption.  Thread.Sleep() does temporarily disable the current thread, so other work must be performed on other threads.  Creating lots of short lived threads may cause a spike in CPU consumption.  Also, creating more threads that your system can handle may cause your application to be unresponsive.

Your application need to manage the number of threads to a reasonable level.  This includes managing batch/scan/query commands that use extra threads from the system thread pool to communicate with all server nodes in parallel.

Batch commands default to calling each node in series, so you should use the default here.  Scans/Queries can be configured to run in series too (See maxConcurrentNodes in ScanPolicy & QueryPolicy).

**Comment by shinkarmv** [comment]
Thank you @BrianNichols for detailed explanation :)

I agree with your comment totally.

Thread.Sleep would not have been root cause of the issue. Since, I have got this in stack trace of one of the blocked thread while analyzing the dump. That is why I suspected as it could be one of the reason.

As you suggested in aforementioned comment, i have cross checked the Query Policy configuration. I did not set the maxConcurrentNodes explicitly. So as per the default, behavior it will issue query request to all nodes. (Currently, I have 3 nodes.)

Regards,
Mukesh
